### PR TITLE
Update font-stack in design tokens

### LINKED ISSUE #128: Synchronize CSS font-stack between DevDot and the design system
### Context

In [this Slack thread](https://hashicorp.slack.com/archives/C01KCU4HDPY/p1648222317499269?thread_ts=1648221234.435329&cid=C01KCU4HDPY) @ashleemboyer and @braydenlove have noticed that the CSS font-stacks for DevDot and HDS differ.

In CoreDevDotLayout ([link](https://github.com/hashicorp/dev-portal/blob/main/src/layouts/core-dev-dot-layout/core-dev-dot-layout.module.css#L6-L13)):
```
--font-body: system-ui, -apple-system, blinkmacsystemfont, 'Segoe UI', roboto, helvetica, arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
--font-display: system-ui, -apple-system, blinkmacsystemfont, 'Segoe UI', roboto, helvetica, arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
--font-monospace: ui-monospace, menlo, consolas, ubuntu mono, monospace;
```

In HDS tokens ([link](https://github.com/hashicorp/design-system/blob/main/packages/tokens/dist/devdot/css/tokens.css#L243-L245)):
```
--token-typography-font-stack-display: -apple-system, BlinkMacSystemFont, SF Pro Display, Segoe UI Display, Ubuntu, sans-serif;
--token-typography-font-stack-text: -apple-system, BlinkMacSystemFont, SF Pro Text, Segoe UI Text, Ubuntu, sans-serif;
--token-typography-font-stack-code: SF Mono, Consolas, Ubuntu Mono, monospace;
```

Discussing among us, we have decided it would be great to align them between the two codebases, so that DevDot can directly use the CSS variables/tokens provided by the design system

### Proposed solution

Below is a list of changes that I am proposing. 

#### Display/Text stack

- **`system-ui`** - **I am proposing not to add it** (so remove it from DevDot) - reason being is that it seems it may cause issues in some cases (apparently it caused issues in WebKit and Blink browsers with non-English languages, see [here](https://infinnie.github.io/blog/2017/systemui.html) and [here](https://markdotto.com/2018/02/07/github-system-fonts/) and [here](https://github.com/twbs/bootstrap/pull/22377)) - these are old issues, and we're not using non-English languages, but since it's just a replacement for `-apple-system` and `BlinkMacSystemFont` and we're already declaring it, there's no real need to add it
- **`-apple-system + BlinkMacSystemFont`** - in both DevDot and HDS, let's just leave them as they are
- **`SF Pro Display/Text`** - These are already in the HDS stack, I suggest we keep them so also DevDot will start to use them
- **`Segoe UI` vs `Segoe UI Display/Text`** - DevDot is using the first one, HDS the second one(s) - I could not find any reference to `Segoe UI Display/Text` online, probably it was me just looking at the design specs in Figma and using the description from there as font name, but is wrong (even the new variable font Segoe UI is just `Segoe UI` - so **I am proposing to change HDS to use `Segoe UI`** as in DevDot
- **`Roboto, Helvetica, Arial`** - they're not present in the HDS stack, so **I am proposing to add them to the HDS stack**
- **`Ubuntu`** - is not present in the DevDot stack, so **I am proposing to add it to the DevDot stack**
- **`Apple Color Emoji, Segoe UI Emoji, Segoe UI Symbol`** - I have not found many references online about why to add these, apart [from this article on CSS Tricks](https://css-tricks.com/snippets/css/system-font-stack/) - I suspect they're not needed anymore, but since they're pretty harmless in any case, **I propose to add them to the HDS stack**

#### Code stack

For the code stack things are much easier:
- **`ui-monospace`** - currently [the support is very low](https://caniuse.com/extended-system-fonts) so **I am proposing to remove it from DevDot** to simplify things
- **`SF Mono`** - this is missing from DevDot, so **I am proposing to add it to DevDot** 
- **`Menlo`** - this is missing from HDS, so **I am proposing to add it to HDS** 

/cc @heatherlarsen @braydenlove @cveigt + @ashleemboyer @hashicorp/design-systems what do you think?

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

This resolves #128 - see the issue for more details on what problem we're trying to solve with this PR

### :hammer_and_wrench: Detailed description

In this PR I have:
- updated the font-stack design tokens in “typography”
  - removed `SF Pro Display/Text` in `Display/Text`
    - as noticed by @heatherlarsen it was useless, `-apple-system, BlinkMacSystemFont` already do the job
  - replaced `Segoe UI Display/Text` with `Segoe UI` in `Display/Text`
  - added `Helvetica, Arial, sans-serif` to the `sans` block in `Display/Text`
    - we initially added also `Roboto` but then decided to remove it ([reference](https://markdotto.com/2018/02/07/github-system-fonts/#:~:text=We%20ended%20up%20removing%20Roboto%20because%20it%20caused%20issues%20in%20some%20distros%20of%20Linux.))
  - removed `Ubuntu` for the `linux` platform
     - reason is that apparently can cause issues ([reference](https://markdotto.com/2018/02/07/github-system-fonts/#:~:text=This%20included%20Roboto%2C%20Oxygen%2C%20Ubuntu%2C%20Cantarell%2C%20Fira%20Sans%2C%20and%20Droid%20Sans.%20Interestingly%20enough%2C%20Linux%20users%20were%20the%20most%20upset%20at%20the%20change%20given%20Linux%20system%20fonts%20seem%20to%20be%20rather%20large%20and%20clunky%2C%20and%20their%20system%20fonts%20so%20inconsistent%20across%20distros.))
  - added explicit emoji support for `Display/Text`
  - replaced `SF Mono` with `ui-monospace` in `Code`
     - notice: this is [the only way to access “SF Mono” via CSS](https://qwtel.com/posts/software/the-monospaced-system-ui-css-font-stack/#:~:text=worth%20expanding%20on%3A-,ui%2Dmonospace,-Future%20system%20monospace) and [it works only on Safari](https://caniuse.com/extended-system-fonts)
     - the previous implementation was wrong and didn't work, it just fell back to the default `monospace` in the browser
  - added `Menlo` to `Code`
- re-generated design tokens in output

### 👀 How to review

👉 Review by files changed

Reviewer's checklist:

- [ ] +1 Percy
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### 🚨 Percy visual regression tests failing

All the visual regression tests in Percy are failing. There is not much we can do. The reasons are multiple:
- The tests for Chrome/Firefox/Edge are likely running on some Linux or Windows machine/emulator, so having changed the font stacks (eg. Segoe + Roboto/Helvetica/Arial) is likely to cause a different font to be used on these OS. 
- Safari was not activated as target in Percy, so we can't see if the tests are failing on Mac/Safari but I did a test in my local machine, and the `Display` and `Body` sans text is exactly the same, while the difference is only on the `Code` monospace (and it's for the reason mentioned above, that the previous implementation was not using the right font declaration).

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202045371056992

### CODE REVIEW

**Comment by MelSumner** [comment]
[question] does design need to sign off on this decision?

**Comment by didoo** [comment]
> [question] does design need to sign off on this decision?

I prefer to have a second eye on this (expecially considered that @heatherlarsen has a good eye for details, and background as front-end developer too) :)

**Comment by heatherlarsen** [comment]
Was doing a bit of research on this as the list is getting pretty lengthy and wanted to share an article I found about [GitHub's explorations with font stacks](https://markdotto.com/2018/02/07/github-system-fonts/#the-stack). This article is a bit dated, but it's still linked on the [Primer website](https://primer.style/design/foundations/typography#font-stack), so seems relevant. I specifically wanted to note that GitHub originally started by including `Roboto, Ubuntu, Cantarell, "Fira Sans", "Droid Sans"`, but later removed them because of feedback they received from Linux users: "Linux system fonts seem to be rather large and clunky, and their system fonts so inconsistent across distros" 

Thankfully we're not including that full list, but what do you think about removing `Roboto, Ubuntu` from the list in an effort to avoid the same issues? Or is this something we want to take away as a potential research topic for the future?

Also wondering if we need to specifically call out "SF Pro Display" and "SF Pro Text"? I may be misinterpreting this, but it seems like we may be able to just rely on `-apple-system, BlinkMacSystemFont`: [Apple Typography Guidelines](https://developer.apple.com/design/human-interface-guidelines/macos/visual-design/typography/)

**Review by ashleemboyer — APPROVED** [review]
This looks good to me! ✨ It'll be awesome to use these in DevDot. 🎉

**Comment by didoo** [comment]
@heatherlarsen if you can have a last look and give me thumbs up, I can merge and release these changes

File: packages/tokens/dist/devdot/css/tokens.css [hunk 1]
```diff
@@ -240,40 +240,40 @@
   --token-surface-overlay-box-shadow: 0 0 0 1px #3b3d4566, 0px 2px 3px 0px #3b3d4580, 0px 12px 24px 0px #3b3d4599;
   --token-focus-ring-action-box-shadow: inset 0 0 0 1px #0c56e9, 0 0 0 3px #5990ff;
   --token-focus-ring-critical-box-shadow: inset 0 0 0 1px #c00005, 0 0 0 3px #dd7578;
-  --token-typography-font-stack-display: -apple-system, BlinkMacSystemFont, SF Pro Display, Segoe UI Display, Ubuntu, sans-serif;
-  --token-typography-font-stack-text: -apple-system, BlinkMacSystemFont, SF Pro Text, Segoe UI Text, Ubuntu, sans-serif;
-  --token-typography-font-stack-code: SF Mono, Consolas, Ubuntu Mono, monospace;
+  --token-typography-font-stack-display: -apple-system, BlinkMacSystemFont, Segoe UI, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji, Segoe UI Symbol;
+  --token-typography-font-stack-text: -apple-system, BlinkMacSystemFont, Segoe UI, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji, Segoe UI Symbol;
```
[the remark this answers is not in the document]
Comment (didoo) [reply packages/tokens/dist/devdot/css/tokens.css:244]: I would prefer not to, if that's OK for you. Reason being that 1) would require some refactoring on our side (not big deal) but potentially from our consumers (not great) but more importantly that, if in the future we want to differentiate for some reasons the two, having two separate tokens will make it seamless (we just release a new version of the package and that's it) while if we have a single token will require to manually check all the use cases where it's used and decide if it's a `text` or a `display` (maybe it's easy, maybe it's a nightmare). 

The cost of maintaining both is zero now (there's not even performance problems). The potential cost of unifying them is unknown (and in the worst case could be pretty high).

[the remark this answers is not in the document]
Comment (heatherlarsen) [reply packages/tokens/dist/devdot/css/tokens.css:244]: Makes sense, thanks for clarifying.

**Review by heatherlarsen — APPROVED** [review]
lgtm!

**Pull request merged**